src/openforms/registrations/contrib/objects_api/tests/test_json_schema.py

```diff
@@ -0,0 +1,211 @@
+    "additionalProperties": False,
+    "properties": {
+        "external": {"$ref": "http://example.com/external-schema.json"},
+    },
+}
+
+
+class IterJsonSchemaTests(SimpleTestCase):
+
+    def test_iter_json_schema_no_refs(self):
+        paths_list = list(iter_json_schema(JSON_SCHEMA_NO_REFS))
+
+        self.assertListEqual(
+            paths_list,
+            [
+                ([], JSON_SCHEMA_NO_REFS),
+                (["complaintDescription"], {"type": "string"}),
+                (["measuredDecibels"], {"type": "array", "items": {"type": "number"}}),
+                (
+                    ["complainant"],
+                    {
+                        "properties": {
+                            "first.name": {"type": "string"},
+                            "last.name": {"type": "string"},
+                        },
+                        "type": "object",
+                    },
+                ),
+                (["complainant", "first.name"], {"type": "string"}),
+                (["complainant", "last.name"], {"type": "string"}),
+            ],
+        )
+
+    def test_iter_json_schema_refs(self):
+        paths_list = list(iter_json_schema(JSON_SCHEMA_REFS))
+
+        self.assertListEqual(
+            paths_list,
+            [
+                ([], JSON_SCHEMA_REFS),
+                (
+                    ["complainant"],
+                    {
+                        "properties": {
+                            "first.name": {"type": "string"},
+                            "last.name": {"type": "string"},
+                        },
+                        "type": "object",
+                    },
+                ),
+                (["complainant", "first.name"], {"type": "string"}),
+                (["complainant", "last.name"], {"type": "string"}),
+                (
+                    ["noisyAddress"],
+                    {
+                        "properties": {
+                            "street": {"type": "string"},
+                        },
+                        "type": "object",
+                    },
+                ),
+                (["noisyAddress", "street"], {"type": "string"}),
+            ],
+        )
+
+    def test_iter_json_schema_nested_refs(self):
+        paths_list = list(iter_json_schema(JSON_SCHEMA_NESTED_REFS))
+
+        self.assertListEqual(
+            paths_list,
+            [
+                ([], JSON_SCHEMA_NESTED_REFS),
+                (
+                    ["complainant"],
+                    {
+                        "properties": {"residence": {"$ref": "#/definitions/address"}},
+                        "type": "object",
+                    },
+                ),
+                (
+                    ["complainant", "residence"],
+                    {"properties": {"street": {"type": "string"}}, "type": "object"},
+                ),
+                (["complainant", "residence", "street"], {"type": "string"}),
+            ],
+        )
```

Comment:
This is still kind of a naive implementation as "deep" references are not resolved until encountered.

This test shows an example of this: the first yielded value has `properties.residence` still unresolved. However, I'm wondering if it should be fully resolved, as I don't think we'll be able to support variables with JSON schemas (yet to be defined, as discussed the `FormVariable` might benefit from a `as_json_schema` method) with deep nested properties